packages/kit/src/runtime/server/page/serialize_data.js

```diff
@@ -46,16 +46,16 @@ export function serialize_data(fetched, filter, prerendering = false) {
 
 	let cache_control = null;
 	let age = null;
-	let vary = false;
+	let varyAny = false;
 
 	for (const [key, value] of fetched.response.headers) {
 		if (filter(key, value)) {
 			headers[key] = value;
 		}
 
 		if (key === 'cache-control') cache_control = value;
-		if (key === 'age') age = value;
-		if (key === 'vary') vary = true;
+		else if (key === 'age') age = value;
+		else if (key === 'vary') varyAny ||= value.trim() === '*';
```

Comment:
I'm slightly confused; `||=` only assigns if LHS is falsy. `varyAny` is always falsy. Can't we just:

```suggestion
		else if (key === 'vary') varyAny = value.trim() === '*';
```

Reply:
I figured this was slightly safer, in case there was one header keyed with `Vary: *`, followed by a header keyed with `Vary: anything else`

Reply:
I don't think it would matter, would it? If you set `Vary: *`, wouldn't all other `Vary` headers become meaningless?

Reply:
I might be over-engineering it 😅 I've updated the code as below, hows this?

```js
else if (key === 'vary' && value.trim() === '*') varyAny = true;
```

Reply:
noice! looks good to me 😁